### PR TITLE
Update nuget config file to use public artifact feed link

### LINKED ISSUE #1395: [Bug]: Nuget.config refers to a private NuGet repository. Unable to build
### What happened?

When opening solution in Visual Studio, authentication is requested for the Nuget package repository referred in the Nuget.config at the root of the repo. 

Renamed the nuget.config in nuget.config.old I was able to restore all the NuGet packages needed to build.

### Version

forked from repo

### What database are you using?

Azure SQL

### What hosting model are you using?

_No response_

### Which API approach are you accessing DAB through?

_No response_

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by abhishekkumams**
## Why make this change?

- Closes #1395 
  - Authentication is requested for the Nuget package repository referred.

## What is this change?

- Updated Nuget.config file to use public nuget repository

## How was this tested?

- Build passed

## Additional Change
**Refactoring** : Moving project properties that applies to all projects to a single file Directory.Build.Props

### CODE REVIEW

**Review by Aniruddh25 — APPROVED** [review]

**Review by Aniruddh25 — CHANGES_REQUESTED** [review]
Please document this in our Contributing docs to modify for building locally. We wont be able to grant permissions to the feed we use to generate our pipeline builds

File: src/Auth/Azure.DataApiBuilder.Auth.csproj [hunk 1]
```diff
@@ -8,14 +8,17 @@
...
   </PropertyGroup>
 
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
     <TreatWarningsAsErrors>True</TreatWarningsAsErrors>
+    <WarningsNotAsErrors>NU1603</WarningsNotAsErrors>
   </PropertyGroup>
 
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|AnyCPU'">
     <TreatWarningsAsErrors>True</TreatWarningsAsErrors>
+    <WarningsNotAsErrors>NU1603</WarningsNotAsErrors>
```
Comment (Aniruddh25) [inline src/Auth/Azure.DataApiBuilder.Auth.csproj:21]: What is needed to avoid getting this warning?

**Review by Aniruddh25 — CHANGES_REQUESTED** [review]
Thanks for finding this solution! Waiting for few more answers.

File: src/Auth/Azure.DataApiBuilder.Auth.csproj (hunk 1 above)
Comment (abhishekkumams) [reply src/Auth/Azure.DataApiBuilder.Auth.csproj:21]: Typically, the package sources do not contain the expected lower bound version. A higher version was used instead, which differs from what the package was authored against. To resolve this we would need to have that lower bound version.

**Review by Aniruddh25 — APPROVED** [review]
LGTM, thanks for driving to fix this!

**Review by seantleonard — DISMISSED** [review]
please check whether we can consolidate this change into a single file.

File: src/Service/Azure.DataApiBuilder.Service.csproj [hunk 2]
```diff
@@ -31,12 +32,15 @@
 
     <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
       <TreatWarningsAsErrors>True</TreatWarningsAsErrors>
+      <WarningsNotAsErrors>NU1603</WarningsNotAsErrors>
```
[the remark this answers is not in the document]
Comment (abhishekkumams) [reply src/Service/Azure.DataApiBuilder.Service.csproj]: moved the repeated code to a single place.

**Review by Aniruddh25 — APPROVED** [review]

**Review by seantleonard — APPROVED** [review]

**Pull request merged**